.travis.yml

```diff
@@ -34,8 +36,11 @@ jobs:
       script:
         # TODO decide whether protoc should be committed or not. If so, we shouldn't do
         # this or we should error if it dirties the repo.
+        - go_deps_sha=$(. ./bin/_tag.sh ; go_deps_sha)
+        - docker build -t gcr.io/linkerd-io/go-deps:${go_deps_sha} -f ./Dockerfile-go-deps .
+        - docker build -t gcr.io/linkerd-io/cni-plugin:travis -f ./cni-plugin/Dockerfile .
         - ./bin/protoc-go.sh
-        - go test -race -v ./...
+        - TAG=travis go test -race -v ./...
```

Comment:
Let's move this to the "integration-test" section, since that only runs after all of the docker images have been built. This means it only runs on commits to the master branch, but I think that's preferable to having to build docker images on every CI run.

I haven't tested it, but something along these lines should work:

```diff
diff --git a/.travis.yml b/.travis.yml
index 00fa96b8..ced3d316 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -160,6 +160,10 @@ jobs:
           version="$(./linkerd version --client --short | tr -cd '[:alnum:]-')"
           ./bin/test-run `pwd`/linkerd linkerd-$version
 
+        - |
+          # Run linkerd-cni integration tests.
+          TAG="$(./linkerd version --client --short)" go test -race ./cni-plugin/test
+
       after_script:
         - |
           # Cleanup after integration test run.
```

Reply:
Yeah, this looks good. I tested out the command and it works.